File: ruby/reverse_string.rb
def reverse_string(str)
  newString = []
  for a in 0..str.length
    newString.unshift(str[a])
  end
  newString.join('')
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 'ih'"
  puts "=>", reverse_string('hi')

  puts

  puts "Expecting: 'ybabtac'"
  puts "=>", reverse_string('catbaby')

  # Don't forget to add your own!
end

# Please add your pseudocode to this file
# And a written explanation of your solution

File: javascript/reverse_string.js
function reverseString(str) {
  let newString = [];
  for (let i = 0; i < str.length; i++) {
    newString.unshift(str[i])
  }
  return newString.join('');
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 'ih'");
  console.log("=>", reverseString("hi"));

  console.log("");

  console.log("Expecting: 'ybabtac'");
  console.log("=>", reverseString("catbaby"));
}

module.exports = reverseString;

// Please add your pseudocode to this file
// And a written explanation of your solution
